File: client/src/main/java/au/com/codeka/warworlds/client/game/world/MyEmpireStarCollection.java
package au.com.codeka.warworlds.client.game.world;

import au.com.codeka.warworlds.client.store.StarCursor;
import au.com.codeka.warworlds.common.proto.Star;

/**
 * Implementation of {@link StarCollection} that represents all of the stars my empire "owns".
 */
public class MyEmpireStarCollection implements StarCollection {
  private StarCursor starCursor;

  public MyEmpireStarCollection() {
    starCursor = StarManager.i.getMyStars();
  }

  @Override
  public int size() {
    return starCursor.getSize();
  }

  @Override
  public Star get(int index) {
    return starCursor.getValue(index);
  }

  @Override
  public int indexOf(long starId) {
    // TODO
    return -1;
  }

  @Override
  public boolean notifyStarModified(Star star) {
    // TODO
    return false;
  }
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/game/world/MyEmpireStarCollection.kt
package au.com.codeka.warworlds.client.game.world

import au.com.codeka.warworlds.client.store.StarCursor
import au.com.codeka.warworlds.common.proto.Star

/**
 * Implementation of [StarCollection] that represents all of the stars my empire "owns".
 */
class MyEmpireStarCollection : StarCollection {
  private val starCursor: StarCursor
  override fun size(): Int {
    return starCursor.size
  }

  override fun get(index: Int): Star {
    return starCursor.getValue(index)
  }

  override fun indexOf(starId: Long): Int {
    // TODO
    return -1
  }

  override fun notifyStarModified(star: Star): Boolean {
    // TODO
    return false
  }

  init {
    starCursor = StarManager.myStars
  }
}